packages/cardano-services/src/Asset/TypeormAssetProvider/TypeormAssetProvider.ts

```diff
@@ -0,0 +1,44 @@
+import { Asset, AssetProvider, Cardano, GetAssetArgs } from '@cardano-sdk/core';
+import { NftMetadataService } from '../types';
+import { TypeormProvider, TypeormProviderDependencies } from '../../util';
+
+interface TypeormAssetProviderDependencies extends TypeormProviderDependencies {
+  nftMetadataService: NftMetadataService;
+}
+
+export class TypeormAssetProvider extends TypeormProvider implements AssetProvider {
+  #dependencies: TypeormAssetProviderDependencies;
+
+  constructor(dependencies: TypeormAssetProviderDependencies) {
+    const { connectionConfig$, entities, logger } = dependencies;
+    super('TypeormAssetProvider', { connectionConfig$, entities, logger });
+
+    this.#dependencies = dependencies;
+  }
+
+  async getAsset({ assetId }: GetAssetArgs): Promise<Asset.AssetInfo> {
+    const assetName = Cardano.AssetId.getAssetName(assetId);
```

Comment:
```suggestion
    const asset = assetRepo.find(...);
    if (!asset) throw new ProviderError(
        ProviderFailure.NotFound,
        undefined,
        `Asset not found '${assetId}'`
      );
```

Reply:
Where is `assetRepo` coming from? Should I do something like this?

```
const assetRepo = queryRunner.manager.getRepository(AssetEntity);
```

Reply:
Yep, you should use `withDataSource` which comes from extending TypeormProvider

Reply:
Fixed on 5f0b649